src/java/org/apache/cassandra/io/sstable/SSTableUniqueIdentifier.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.io.sstable;
+
+import java.nio.ByteBuffer;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+/**
+ * Represents a unique identifier in the sstable descriptor filename.
+ * This ensures each sstable file uniqueness in the entire Cassandra cluster.
+ * <p>
+ * A new implementation must adhere to the following invariants:
+ * - Must be locally sortable - that is, the comparison must reflect the comparison of generation times of identifiers
+ * generated on the same node
+ * - String representation must *not* include the '-' character see {@link Descriptor#filenameSplitter}
+ * - must be case-insensitive because the sstables can be stored on case-insensitive file system
+ * <p>
+ */
+public interface SSTableUniqueIdentifier
```

Comment:
What would you think about naming this `SSTableId` for the sake of brevity, as we do with [`TableId`](https://github.com/apache/cassandra/blob/eb0631646f824a4998556386ebd979ea31874c0f/src/java/org/apache/cassandra/schema/TableId.java)? 

I think that the JavaDoc is clear enough to don't let us miss that this is an unique identifier, and the shorter name seems to help a bit with the long signatures and method calls. I gave it a go in [this commit](https://github.com/adelapena/cassandra/commit/edae13c7a88491e6383fda13ae246392e943749a), if you want to see how it would look like.

Reply:
I'm ok with this idea 👍🏻